packages/flutter/test/rendering/wrap_test.dart

```diff
@@ -12,17 +12,146 @@ void main() {
+      _intrinsicHeightDefaultTest(children, 245, 165, isMax: false);
+      _intrinsicHeightDefaultTest(children, 250, 80, isMax: false);
+      _intrinsicHeightDefaultTest(children, 80, 250, isMax: false);
+      _intrinsicHeightDefaultTest(children, 79, 250, isMax: false);
+    },
+  );
+
+  test(
+    'Compute intrinsic width test',
+    () {
+      final List<RenderBox> children = <RenderBox>[
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+      ];
+
+      _intrinsicWidthDefaultTest(children, 245, 165);
+      _intrinsicWidthDefaultTest(children, 250, 80);
+      _intrinsicWidthDefaultTest(children, 80, 250);
+      _intrinsicWidthDefaultTest(children, 79, 250);
+      _intrinsicWidthDefaultTest(children, 245, 165, isMax: false);
+      _intrinsicWidthDefaultTest(children, 250, 80, isMax: false);
+      _intrinsicWidthDefaultTest(children, 80, 250, isMax: false);
+      _intrinsicWidthDefaultTest(children, 79, 250, isMax: false);
+    },
+  );
+}
+
+void _intrinsicWidthDefaultTest(
+  List<RenderBox> children,
+  double availableHeight,
+  double expectedWidth, {
+  Axis direction = Axis.vertical,
+  double spacing = 5.0,
+  double runSpacing = 5.0,
+  bool isMax = true,
+}) {
+  final RenderWrap renderWrap = _prepareWrapForComputeIntrinsicSize(
+    children,
+    spacing: spacing,
+    runSpacing: runSpacing,
+    direction: direction,
+  );
+  expect(
+    isMax
+        ? renderWrap.computeMaxIntrinsicWidth(availableHeight)
+        : renderWrap.computeMinIntrinsicWidth(availableHeight),
+    equals(expectedWidth),
+  );
+
+  // clear for available reuse children in test
+  renderWrap.removeAll();
+}
+
+void _intrinsicHeightDefaultTest(
+  List<RenderBox> children,
+  double availableWidth,
+  double expectedHeight, {
+  Axis direction = Axis.horizontal,
+  double spacing = 5.0,
+  double runSpacing = 5.0,
+  bool isMax = true,
+}) {
+  final RenderWrap renderWrap = _prepareWrapForComputeIntrinsicSize(
+    children,
+    spacing: spacing,
+    runSpacing: runSpacing,
+    direction: direction,
+  );
+  expect(
+    isMax
+        ? renderWrap.computeMaxIntrinsicHeight(availableWidth)
+        : renderWrap.computeMinIntrinsicHeight(availableWidth),
+    equals(expectedHeight),
+  );
+
+  // clear for available reuse children in test
+  renderWrap.removeAll();
+}
+
+RenderWrap _prepareWrapForComputeIntrinsicSize(
+  List<RenderBox> children, {
+  Axis direction = Axis.horizontal,
+  double spacing = 5.0,
+  double runSpacing = 5.0,
+}) {
+  final RenderWrap renderWrap = RenderWrap();
+
+  // ignore: prefer_foreach
```

Comment:
remove the ignore and use foreach

Reply:
@goderbauer when i use foreach i got a warning https://dart-lang.github.io/linter/lints/avoid_function_literals_in_foreach_calls.html